### PR TITLE
- Added `MaxPermSize=256m` for the SparkWikipediaClustering

### PR DESCRIPTION
**Pull request by chtyim**
- Improve UncaughtExceptionHandler Error logging.

### CODE REVIEW

File: cdap-common/src/main/java/co/cask/cdap/common/logging/common/UncaughtExceptionHandler.java [hunk 1]
```diff
@@ -27,6 +27,15 @@
 
   @Override
   public void uncaughtException(Thread t, Throwable e) {
+    // If the Throwable is an Error, the system most likely is in unusable state.
+    // Hence we try our best effort to log the message first to at least get some information,
+    // followed by logging the stacktrace (in OOM case, it may fail to get the stacktrace)
+    if (e instanceof Error) {
+      LOG.error("Uncaught error in thread {}", t, e.getMessage());
```
Comment (anwar6953) [inline cdap-common/src/main/java/co/cask/cdap/common/logging/common/UncaughtExceptionHandler.java]: Don't you need another `{}` in this log message?

Comment (chtyim) [reply cdap-common/src/main/java/co/cask/cdap/common/logging/common/UncaughtExceptionHandler.java]: right.

**Comment by anwar6953** [comment]
LGTM

**Comment by anwar6953** [comment]
https://builds.cask.co/browse/CDAP-RUT995-3

**Pull request merged**